src/main/java/no/uib/inf112/core/map/MapInteractOnUser.java

```diff
@@ -36,18 +38,54 @@ public boolean scan(Collection<Entity> entitiesOnMap) {
      */
     private void registerSpecialTiles(@NotNull Collection<Entity> entitiesOnMap) {
         ArrayList<MapAction> queue = new ArrayList<>(); //Not a queue but using it as a queue
-        for (Entity entity : entitiesOnMap) {
-            MapAction mapAction = getAction(entity);
-            if (mapAction != null) {//And mapaction is some special tile
-                queue.add(mapAction);
+
+        for (Player player : RoboRally.getPlayerHandler().getPlayers()) {
```

Comment:
This call for all players should be different, either the scan method should be given a list of players or the list of players should be the same for the entire cycle (use the RoboRally.getPlayerHandler().getPlayers() in the scan method)

Reply:
The MapAction interface as it is now doesn't really make sense to use for the registration of flags. I can move the code for flag registration out into its own class, but it can't really use the methods described by the interface so I don't see the point in implementing it?

Reply:
i see your point, we can do some refactoring at a later stage to make it more fitting.